### PR TITLE
Adjust release doc

### PR DESCRIPTION
**Pull request by olukas**
Couple of adjustments in `Major and Minor Releases` + I unified versions in examples for `Patch Releases`. @Holmistr please review whether it's reflect conclusions from our meeting.

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -73,13 +73,13 @@ The documentation site will now be built, including content from your tagged ver
 
 This section guides you through the steps for releasing documentation for major and minor releases.
 
-. Create a `\{version\}.z` branch from the `master` branch, where `\{version\}` is the version of IMDG that you are releasing. For example, if you're working on content for version 5.0, create a branch named `5.0.z`.
+. Work on the content for a new Major or Minor release on the `master` branch (there is an exception - if you work on content for the new Minor branch which is lower than current Major version then you need to create `\{version\}.y` branch. For example if `5.0` was released and you plan to add content for `4.3` then you need to create `4.y` branch and work on it. Then use this branch in following text instead of `master` branch).
```
Comment (JakeSCahill) [inline README.adoc:76]:
```suggestion
. If you're writing content for a new major or minor release, make your changes to the `master` branch.

. If you're writing content for a new minor version of a previous major release, create a `\{version\}.y` branch and work on that. For example if version 5.0 was already released and you plan to add content for version 4.3 then you need to create a branch named`4.y`.
```

File: README.adoc [hunk 2]
```diff
@@ -73,13 +73,13 @@ The documentation site will now be built, including content from your tagged ver
...
 +
 TIP: For guidance on writing content, see the {url-contribute}[contribution guidelines].
 
-. When you are ready to release, create a new release branch from the `.z` branch and name it with the full version number that you are releasing. For example, if you are releasing version 5.0, name the branch 5.0.
+. When you are ready to release, create a new release branch from the `master` branch and name it with the full version number that you are releasing. For example, if you are releasing version 5.0, name the branch 5.0.
```
Comment (JakeSCahill) [inline README.adoc:80]:
```suggestion
. When you are ready to release, create a new release branch from the `master` or `.y` branch and name it with the full version number that you are releasing. For example, if you are releasing version 5.0, name the branch 5.0.
```

File: README.adoc [hunk 3]
```diff
@@ -88,13 +88,13 @@ version: 5.0
 
 . Tag the latest commit in your release branch with a `v` followed by the version number.
 
-. If you have made any changes since creating your release branch, merge those changes back into the `.z` branch.
+. Create a `\{version\}.z` branch from release branch. It ensures that all changes made since creating your release branch will be applied to the new `.z` branch. For example `5.0.z` branch.
```
Comment (JakeSCahill) [inline README.adoc:93]: So we create a `.z` branch to allow others to add content to future path releases of that minor version? Should we explain that here?

**Review by Holmistr — APPROVED** [review]

**Review by JakeSCahill — CHANGES_REQUESTED** [review]
Some suggestions

File: README.adoc (hunk 3 above)
Comment (olukas) [reply README.adoc:93]: Basically we create `.z` branch in the time when given Major/Minor release if tagged (i.e. it the time it's released). Generally `.z` branches are used for some smaller fixes of given branches. I'm not sure how we should explain it there, feel free to suggest something.

Comment (JakeSCahill) [inline README.adoc:95]:
```suggestion
. To make it easier to amend your release for future patch versions or to fix typos,  create a `\{version\}.z` branch from your release branch. For example if your release branch is named `5.0`, create a branch named `5.0.z`.
```

**Review by JakeSCahill — APPROVED** [review]
Thanks 😄

**Pull request merged**